packages/muelu/src/Utils/MueLu_Utilities.cpp

```diff
@@ -248,4 +256,43 @@ bool IsParamValidVariable(const std::string& name)
     }
   }
 
+
+   // Generates a communicator whose only members are other ranks of the baseComm on my node
+    Teuchos::RCP<const Teuchos::Comm<int> > GenerateNodeComm(RCP<const Teuchos::Comm<int> > & baseComm, int &NodeId) {
+#ifdef HAVE_MPI
+       int numRanks = baseComm->getSize();       
+       if(numRanks == 1) {NodeId = baseComm->getRank(); return baseComm;}
+
+       // Get an integer from the hostname
+       char hostname[MPI_MAX_PROCESSOR_NAME];
+       int len;
+       MPI_Get_processor_name(hostname,&len);
+       struct hostent * host = gethostbyname(hostname);
+       int myaddr = (int) htonl(inet_network(inet_ntoa(*(struct in_addr *)host->h_addr)));
+       
+       // All-to-all exchange of address integers
+       std::vector<int> addressList(numRanks);
+       Teuchos::gatherAll(*baseComm,1,&myaddr,numRanks,&addressList[0]);
+
+       // Sort!
+       std::sort(addressList.begin(),addressList.end());
+
+       // Find the number of node-buddies I have

```

Comment:
This comment misled me.  The code below is identifying the node ID of the current MPI rank, counting up from zero, isn't it?  All MPI ranks on a node should calculate the same node ID.

Reply:
Oh, yeah, that comment is not correct.